orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/OperationsController.groovy

```diff
@@ -77,15 +77,45 @@ class OperationsController {
 
   @RequestMapping(value = "/orchestrate", method = RequestMethod.POST)
   Map<String, Object> orchestrate(@RequestBody Map pipeline, HttpServletResponse response) {
-    parsePipelineTrigger(executionRepository, buildService, pipeline)
-    Trigger trigger = objectMapper.convertValue(pipeline.trigger, Trigger)
+    if (pipeline.plan) {
+      parseAndValidatePipeline(pipeline)
+      log.info('Not starting pipeline (plan: true): {}', value("pipelineId", pipeline.id))
+      return pipeline
+    }
+
+    Throwable pipelineError = null
+    try {
+      parseAndValidatePipeline(pipeline)
```

Comment:
Seems like the logic could be cleaned up here such that we only need to call `parseAndValidatePipeline(pipeline)` once and then take some follow-up action depending `pipeline.plan` ?

Reply:
Yeah, that's reasonable, I'll clean this up